### DISCUSSION TITLE
How to extend cockpit and integrate it into my project

### DISCUSSION THREAD

**Discussion by zhangziyang960** [cockpit-project/cockpit#19463] [Q&A] [answered]
I am a front-end engineer and have received a request to study how to extend the functionality of Cockpit (add pages) and integrate it into my project. What should I do instead?

**Comment by martinpitt** [comment, marked as answer]
Do instead of what?

If you are looking for some documentation how to create your own Cockpit page, please have a look here: [Contribute → Further reading](https://cockpit-project.org/external/wiki/Contributing.html#further-reading)

**Comment by equ1pto** [comment]
@martinpitt  Don't be so nice to this "engineer" - they know damn well where the docs are...
![image](https://github.com/cockpit-project/cockpit/assets/29682585/8ad94c75-a7af-4559-ac9b-4c0f24216586)
"What should i do INSTEAD!?" $hoot - quit your "job" OR wait until they let you go...OR just do the WORK *, maybe?
